osu.Game/Rulesets/Objects/Pooling/HitObjectEntryManager.cs

```diff
@@ -105,16 +100,16 @@ public bool TryGet(HitObject hitObject, [MaybeNullWhen(false)] out HitObjectLife
         /// </summary>
         private void onDefaultsApplied(HitObject hitObject)
         {
-            if (!childrenMap.Remove(hitObject, out var childEntries))
+            if (!entryMap.TryGetValue(hitObject, out var entry))
```

Comment:
I'm not sure on the performance considerations of this, but I think it's probably fine to go with until that is raised as a concern.